Tests/Integration/ClassCIntegrationTests.cs

```diff
@@ -173,10 +173,6 @@ public async Task When_OTAA_Join_Then_Sends_Upstream_DirectMethod_Should_Send_Do
                     savedAppSKey = AppSessionKey.Parse(t[TwinProperty.AppSKey].Value);
                     savedNwkSKey = NetworkSessionKey.Parse(t[TwinProperty.NwkSKey].Value);
                     savedDevAddr = t[TwinProperty.DevAddr];
-
-                    Assert.NotNull(savedAppSKey);
-                    Assert.NotNull(savedNwkSKey);
-                    Assert.NotEmpty(savedDevAddr);
```

Comment:
A general comment about the issue just to double check that I understand it correctly: changing these assertions for me causes the test to fail as it should. I think the problem with these assertions is that they make it hard to debug failures because they are reported on the call site (i.e. Line 210 instead of here), right?

Reply:
I was able to reproduce on my machine and the CI (run https://github.com/Azure/iotedge-lorawan-starterkit/actions/runs/1744091670 on commit 310959f9ca653c7170e9ed3fe1a8f680f080feb2) that changing an assertion does not fail the test. Do we want to sync offline why it fails on your machine?